### ISSUE TITLE
bug: ionic.bundle.js causes ng-click event to fire 2x

### ISSUE THREAD

**Issue by shredmer** [driftyco/ionic#4345]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>ios</span> <span ionic-platform-version>8</span>  <span ionic-webview>webview</span>

<span ionic-description>http://play.ionic.io/app/2fc7300bf4a6

All ng-click events, with an alert, are firing twice. Clicking the button fires the alert once, the moment you click/touch off of the button anywhere else on the screen the alert fires again.

This bug is not produced in the browser or on android. It is however produced on the ios emulator and a real ios device.

UPDATE: As far as I can tell this is directly tied to using alert() or $window.alert() with a button or link. Removing the alert and using console.log alone, results in no additional click. Including the alert fires the entire event again, not just alert alone.

WORK-AROUND: By replacing the alert with an alert using the cordovaDialog plugin, the issue does not occur.

This is a link to the discussion on the forum: http://forum.ionicframework.com/t/ionic-bundle-js-causes-ng-click-event-to-fire-multiple-times/32008</span>

<span is-issue-template></span>

**Comment by jeveloper** [comment]
That could be pretty bad if integrating analytics

**Comment by shredmer** [comment]
Turns out this is most likely tied to alert() and $window.alert(). By replacing this alert with a cordovaDialog, the bug does not occur.

**Comment by mahnuh** [comment]
I noticed this behaviour, if the button that calls the alert is directly beneath the alert's OK button. So basically on closing the alert you press the button again and the alert pops up again.
Can you confirm that?

**Comment by shredmer** [comment]
This may also happen. However I have noticed that the button does not have to be directly beneath the OK button. Simply clicking the button once fires the alert, then clicking ok does not fire the alert again. It is once the user clicks/touches anywhere on the device the alert will fire again.

**Comment by mhartington** [comment]
So far, I've only been able to replicate this issue with an alert. I switched the demo's alert out for a console.log, and wasn't able to replicate it. Even using the ionic popups, the issue is not present.

What I think is happening is the system alert is causing the issue. Can you guys confirm this?

**Comment by shredmer** [comment]
Yes that is the same conclusion I ran into. The issue occurs using alert() or $window.alert()

**Comment by mhartington** [comment]
This makes sense. Using window.alert can pause any js going around and prevent proper execution. I did notice that why I switched to the cordova dialogs plugin or ionic's popups, that the issue was gone.

This seems to be the way the browser works, though I'm curious as to why it's not visible on desktop. Not sure what we can do here. But my advice would be to not use window.alerts anyways. Not only to avoid this issue, but not sure the nasty `index.htm` at the top of the alert. Either the ionic popups or cordova dialogs make for a good alternative.

**Comment by shredmer** [comment]
Agreed.

In my case, I was using the alert for debugging purposes and was ignoring the double-fire until it actually was causing problems with my project. Since I did not know what was causing the issue, and being new to Ionic, I ended up spending a good amount of time trying to figure out what was going on. 

From my perspective at the time, any ng-click related event was being fired twice due to ionic.bundle.js, only to realize I had an alert() in every button which was causing each button to independently fire twice.

Anyways, if anyone else runs into this hopefully they will find this thread before pulling their hair out as I did haha
